routers/api/v1/repo/release.go

```diff
@@ -62,42 +61,30 @@ func CreateRelease(ctx *context.APIContext, form api.CreateReleaseOption) {
 		ctx.Status(404)
 		return
 	}
-	tag, err := ctx.Repo.GitRepo.GetTag(form.TagName)
+	rel, err := models.GetRelease(ctx.Repo.Repository.ID, form.TagName)
 	if err != nil {
-		ctx.Error(500, "GetTag", err)
+		if models.IsErrReleaseNotExist(err) {
+			ctx.Status(404)
+			return
+		}
+		ctx.Handle(500, "GetRelease", err)
 		return
 	}
-	commit, err := tag.Commit()
-	if err != nil {
-		ctx.Error(500, "Commit", err)
+
+	if !rel.IsTag {
+		ctx.Status(409)
```

Comment:
Typo?

Reply:
If it is type than it was also previously I kept previous behaviour, it was returning status 409 when release already existed

Reply:
I think it is not a typo: https://github.com/go-gitea/gitea/blob/93a1de48428c2f89502cf6bba80f3976c6f5839f/routers/api/v1/repo/release.go#L97